doc/directory-structure.md

```diff
@@ -0,0 +1,183 @@
+├── exporters/    # Build system export utilities
+├── external/     # External dependencies
+├── fuzz/         # Fuzzing tools and configuration
+├── include/      # Public header files
+├── ms/           # Microsoft Windows specific files
+├── os-dep/       # Operating system dependent code
+├── providers/    # Provider implementations (FIPS, etc.)
+├── ssl/          # SSL/TLS implementation
+├── test/         # Test suite
+├── tools/        # Various utility tools
+├── util/         # Build-related utilities
+└── VMS/          # OpenVMS specific files
+```
+
+## Root-Level Files
+
+The repository root contains many important files:
+
+* `README.md` - The main README with project overview and links to documentation
+* `INSTALL.md` - Detailed installation instructions for all platforms
+* `Configure` - The main build configuration script
+* `config` - A wrapper around Configure for Unix platforms
+* `LICENSE.txt` - The Apache 2.0 license that applies to OpenSSL 3.0+
+* `CHANGES.md` - A comprehensive changelog
+* `NEWS.md` - A curated list of major changes in each release
+* `VERSION.dat` - Contains the current version information
+* `*.pc.in` and `configdata.pm.in` - Templates used during build configuration
+
+Various platform-specific notes are also at the root level:
+* `NOTES-*.md` - Platform-specific information (UNIX, WINDOWS, etc.)
+* `README-*.md` - Additional READMEs for specific components 
+  (ENGINES, FIPS, etc.)
+
+## Major Directories
+
+### apps/
+
+Contains the OpenSSL command-line application source code. 
+This is what you're using when you run the `openssl` command.
+
+Notable components:
+* The `openssl s_client` tool supports QUIC via the `-quic` flag 
+  (since version 3.2.0)
+* The `openssl s_server` tool does NOT support QUIC directly
+
+### crypto/
+
+The heart of OpenSSL containing implementations of cryptographic algorithms. 
+Inside you'll find subdirectories for different cryptographic functions. 
+This is the code that ends up in `libcrypto`.
+
+* `crypto/aes/` - AES encryption implementation
+* `crypto/bn/` - Big Number arithmetic library
+* `crypto/evp/` - High-level cryptographic API
+* `crypto/rsa/`, `crypto/ec/`, etc. - Various algorithm implementations
+
+### demos/
+
+Example code demonstrating how to use OpenSSL APIs:
+
+* `demos/quic/` - QUIC protocol examples, including a dedicated server 
+  implementation
+  * `demos/quic/server/` - Contains a standalone QUIC server application 
+    that should be used instead of `s_server` for QUIC functionality
+  * Usage: `./demos/quic/server/server <port> <certificate-file> <key-file>`
+
+### doc/
+
+Documentation organized in multiple formats:
+
+* `doc/man1/` - Manual pages for command-line tools
+* `doc/man3/` - Manual pages for API functions
+* `doc/man5/` - Manual pages for file formats
+* `doc/man7/` - Manual pages for overviews and concepts
+* `doc/internal/` - Developer documentation
+
+### include/
+
+Header files that are installed along with OpenSSL:
```

Comment:
Well - only `include/openssl` gets "installed" (i.e. when you run `make install`). The other stuff is just internal and does not form part of the public API